File: src/main/java/org/jukeboxmc/block/BlockPortal.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemPortal;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockPortal extends Block {

    public BlockPortal() {
        super( "minecraft:portal" );
    }

    @Override
    public ItemPortal toItem() {
        return new ItemPortal();
    }

    @Override
    public BlockType getType() {
        return BlockType.PORTAL;
    }

    public void setPortalAxis( PortalAxis portalAxis ) {
        this.setState( "portal_axis", portalAxis.name().toLowerCase() );
    }

    public PortalAxis getPortalAxis() {
        return this.stateExists( "portal_axis" ) ? PortalAxis.valueOf( this.getStringState( "portal_axis" ) ) : PortalAxis.X;
    }

    public enum PortalAxis {
        UNKNOWN,
        X,
        Z
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockPortal.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.Portal
import org.jukeboxmc.api.block.data.PortalAxis
import org.jukeboxmc.server.block.JukeboxBlock

class BlockPortal(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), Portal {

    override fun canPassThrough(): Boolean {
        return true
    }

    override fun getPortalAxis(): PortalAxis {
        return PortalAxis.valueOf(this.getStringState("portal_axis"))
    }

    override fun setPortalAxis(value: PortalAxis): Portal {
        return this.setState("portal_axis", value.name.lowercase())
    }
}